krcal/core/fit_lt_functions_test.py

```diff
@@ -150,3 +153,78 @@ def test_pars_from_fcs_warns_for_invalid_fits(caplog):
 
     with warns(UserWarning, match="fit did not succeed, returning NaN"):
         pars_from_fcs([fitcol])
+
+
+@given(integers(min_value = 1e4,
+                max_value = 1e6),
+       floats  (min_value = 1e+4,
+                max_value = 1e+6),
+       floats  (min_value = 2000,
+                max_value = 1e+4),
+       integers(min_value = 10,
+                max_value = 60),
+       floats  (min_value = 100,
+                max_value = 500))
+def test_fit_lt_unbined_change_zrange(Nevt, e0, lt, nbins_z, max_z):
+    """
+    This tests checks that the fit parameters change if range_z
+    do so, i.e. this variable is being used in fit_lt_unbined.
+    Before, this fact didn't happen, therefore the test would fail.
+    """
+    std = 0.05 * e0
+    z, es = energy_lt_experiment(Nevt, e0, lt, std, zmax=max_z)
+
+    z_range_1 = (0, max_z)
+    z_range_2 = (0, max_z/2)
+    _, _, pars_1 = fit_lifetime_unbined(z, es, nbins_z, z_range_1);
+    _, _, pars_2 = fit_lifetime_unbined(z, es, nbins_z, z_range_2);
+
+    lt_1 , e0_1  = pars_1.par
+    ltu_1, e0u_1 = pars_1.err
+    lt_2 , e0_2  = pars_2.par
+    ltu_2, e0u_2 = pars_2.err
+
+    assert lt_1  != lt_2
```

Comment:
I have impression this is doomed to fail with hypothesis, eventually it will draw points that will break this